pytest/tests/loadtest/locust/common/base.py

```diff
@@ -102,6 +103,35 @@ def sender_account(self) -> Account:
         """
 
 
+class FunctionCall(Transaction):
+
+    def __init__(self,
+                 sender: Account,
+                 receiver_id: str,
+                 method: str,
+                 balance: int = 0):
+        super().__init__()
+        self.sender = sender
+        self.receiver_id = receiver_id
+        self.method = method
+        self.balance = balance
+
+    @abc.abstractmethod
+    def args(self) -> dict:
+        """
+        Function call arguments to be serialized and sent with the call.
+        """
+
+    def sign_and_serialize(self, block_hash) -> bytes:
+        return transaction.sign_function_call_tx(
+            self.sender.key, self.receiver_id, self.method,
+            json.dumps(self.args()).encode('utf-8'), 300 * TGAS,
+            int(self.balance), self.sender.use_nonce(), block_hash)
```

Comment:
Given that we expect the balance to be an `int` here, maybe we should convert it to `int` either in `FunctionCall` constructor or even at the call sites.

Reply:
Yeah I think ideally it would be at the constructor function signature, forcing all call sites to use `int`. But it's to easy to use `1e26` or some other form of float, since we don't enforce type hints.
=> I've updated all call sites, but also added a defensive cast inside the constructor.